connector_helpscout/models/helpscout_conversation/importer.py

```diff
@@ -0,0 +1,131 @@
+        conversation = self.env['project.task'].search([
+            ('code', '=', str(record.number)),
+        ])
+
+        if conversation:
+            return {'odoo_id': conversation.id}
+
+    @mapping
+    def helpscout_folder_id(self, record):
+        folder_external_id = "%d,%d" % (
+            record.mailbox.id,
+            record.folder_id,
+        )
+        folder_binding = self.env['helpscout.helpscout.folder'].search([
+            ('external_id', '=', folder_external_id)
+        ])
+        folder = folder_binding.odoo_id
+        if folder:
+            return {'helpscout_folder_id': folder.id}
+
+    @mapping
+    def partner_id(self, record):
+        customer = self.env['helpscout.customer'].search([
+            ('external_id', '=', record.customer.id)
+        ])
+        partner = customer.odoo_id
+        return {'partner_id': partner.id}
+
+    @mapping
+    def project_id(self, record):
+        mailbox = self.env['helpscout.mailbox'].search([
+            ('external_id', '=', record.mailbox.id)
+        ])
+        project = mailbox.odoo_id
+        if project:
+            return {'project_id': project.id}
+
+    @mapping
+    def stage_id(self, record):
+        stage = self.env.ref('connector_helpscout.status_%s' % record.status)
+        if stage:
+            return {'stage_id': stage.id}
+
+    @mapping
+    def tag_ids(self, record):
+        tags = self.env['project.tags'].search([
+            ('name', 'in', record.tags)
+        ])
+        if tags:
+            tag_ids = tags.mapped('id')
+            return {'tag_ids': [(6, 0, tag_ids)]}
+
+    @mapping
+    def user_id(self, record):
+        try:
+            owner = self.env['helpscout.user'].search([
+                ('external_id', '=', record.owner.id)
+            ])
+        except AttributeError:
+            # in case a user/owner has not been assigned to the ticket
+            return {'user_id': False}
+        user = owner.odoo_id
+        return {'user_id': user.id}
+
+
+class HelpScoutConversationImporter(Component):
+    """Import one HelpScout record."""
+    _name = 'helpscout.record.importer.conversation'
+    _inherit = 'helpscout.importer'
+    _apply_on = 'helpscout.conversation'
+
+    def _import_dependencies(self):
+        """
+        Import dependencies for HelpScout conversation.
+
+        Folders are found using the Mailbox id, not the folder id.
+        Tags can only be retrieved as a complete list.
+        """
+        self.env['helpscout.helpscout.folder'].import_batch(
+            self.backend_record,
+            {'mailbox_id': self.helpscout_record['mailbox']['id']},
+        )
+        self.env['helpscout.tag'].import_batch(self.backend_record)
+
+    def _after_import(self, binding):
+        binding_model = self.env['helpscout.thread'].with_context(
+            tracking_disable=True,
+        )
+        thread_ids = [
+            binding_model.import_direct(self.backend_record, thread).id
```

Comment:
You should always just work with recordsets instead of mapping to `ids`. So this would be `threads = [..]` & the below `.browse` should be removed. Or was this for a specific reason?

Reply:
The idea here was to get a recordset for the res_id/model write, since I just have a list of newly-imported thread records at this point. Is there a better way to do this?

Reply:
Well I think you're doing this fine, it's just that you should ALWAYS be working with recordsets until the last possible moment. In this case, you're creating a list of IDs that could easily be obtained with `recordset.ids`

Reply:
But I don't have a recordset until I get it with `.browse`, just a list of individual records that were just imported. Or am I missing something obvious?

Reply:
AFAIK the binding record is returned from `importer.run`, which is returned by `import_direct` & alluded to here by the fact that you are performing a `.id` on the result:
```
binding_model.import_direct(self.backend_record, thread).id
```

So instead I am recommending:

```
thread_bindings = sum(
    binding_model.import_direct(self.backend_record, thread)
    for thread in self.helpscout_record['threads']
)
```

or something like that. Then:

```
thread_bindings.write({
    'model': ...
```

Reply:
Ah, thanks. I follow you on working with recordsets when possible, I was just trying to get something I could actually call `.write` on. I'll try `sum`.

Reply:
if sum doesn't work, just do like this:

```
thread_bindings = binding_model
for thread in self.helpscout_record['threads']:
    thread_bindings += import
```